packages/veraswap-test-duster/src/pages/api/[chainId]/[address].ts

```diff
@@ -126,55 +52,59 @@ export default async function handler(
         transport: http(),
     });
 
-    const balance = await publicClient.getBalance({
-        address: receiver,
-    });
+    const tokens = Object.values([...localMockTokens, ...testnetMockTokens]).filter(
+        (token) => token.chainId === chainIdParsed
+    ) as Token[];
 
-    const min = parseEther("0.0005");
 
-    if (balance < min) {
-        const hash = await walletClient.sendTransaction({
-            to: address,
-            value: (min * 150n) / 100n,
-        });
-        await publicClient.waitForTransactionReceipt({ hash });
-        console.log(`Sent 0.00075 ETH to ${address}`);
-    }
+    const calls = await getDustAccountCalls({
+        account: receiver,
+        client: publicClient as PublicClient<Transport, Chain>,
+        tokens,
+    });
 
-    const tokensForChain = Object.values(TOKENS).filter(
-        (token) => token.chainId === chainIdParsed
-    ) as Array<{ address: Address; decimals: number; symbol: string }>;
-
-    for (const token of tokensForChain) {
-        const balance = await publicClient.readContract({
-            address: token.address,
-            abi: MockERC20.abi,
-            functionName: "balanceOf",
-            args: [receiver],
-        });
-
-        const target = parseUnits("50", token.decimals);
-        if (balance > target) {
-            continue;
-        }
 
-        try {
+    if (calls.length === 0) {
+        res.status(200).send("Nothing to dust");
+        return;
+    }
+    if (calls.length === 1) {
+        const call = calls[0];
+        if (!call.data) {
```

Comment:
This is off, the call is already encoded. There should be no need for this `if/else` condition and a simple `sendTransaction` should suffice.